### PR TITLE
Return `FrozenList` for `Index.names`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: #16046 

This PR returns `FrozenList` for `Index.names` instead of `tuple`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/core/_base_index.py [hunk 1]
```diff
@@ -344,7 +344,7 @@ def names(self):
         """
         Returns a tuple containing the name of the Index.
```
Comment (bdice) [inline python/cudf/cudf/core/_base_index.py:345]: Make this match the pandas docstring, or make the type correct in the summary.
```suggestion
        Returns a FrozenList containing the name of the Index.
```

File: python/dask_cudf/dask_cudf/io/parquet.py [hunk 2]
```diff
@@ -316,7 +316,7 @@ def read_partition(
 
             if index and (index[0] in df.columns):
                 df = df.set_index(index[0])
-            elif index is False and df.index.names != (None,):
+            elif index is False and tuple(df.index.names) != (None,):
```
Comment (bdice) [inline python/dask_cudf/dask_cudf/io/parquet.py:319]: Can you do direct comparison of a FrozenList and a list? That might be more “natural” if it works.
```suggestion
            elif index is False and df.index.names != [None]:
```

**Review by bdice — APPROVED** [review]
Approving with two small requests.

**Review by wence- — APPROVED** [review]

**Pull request merged**